### PR TITLE
Added FMW Integration test with Dynamic Cluster using MII

### PR DESCRIPTION
**Pull request by hzhao-github**
Added FMW Integration test with Dynamic Cluster using MII

Jenkin Result
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4187 (parallel)

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/4196 (sequential)

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:1]: Modify the class name to ItFmwDynamicClusterMiiDomain.java to best represent the usecae

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    installAndVerifyOperator(opNamespace, domainNamespace);
+  }
+
+  /**
+   * Create a basic JRF dynamic cluster model in image domain.
+   * Verify Pod is ready and service exists for both admin server and managed servers.
+   * Verify EM console is accessible.
+   */
+  @Test
+  @DisplayName("Create JRF Dynamic Domain model in image")
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:143]: Change the Display Name  to "Create FMW Dynamic with Dynamic Cluster using model in image"

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+  public void testJrfDynamicDomainModelInImage() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:144]: Modify the test name to testFmwDynamicClusterDomainModelInImage

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    // Create the repo secret to pull the image
+    // this secret is used only for non-kind cluster
+    createOcirRepoSecret(domainNamespace);
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    assertDoesNotThrow(() -> createSecretWithUsernamePassword(
+        adminSecretName,
+        domainNamespace,
+        "weblogic",
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:160]: Can you use the TestConstant for user/password

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+                    .introspectorJobActiveDeadlineSeconds(600L)));
+
+    createDomainAndVerify(domain, domainNamespace);
+  }
+
+  /**
+   * Verify Pod is ready and service exists for both admin server and managed servers.
+   * Verify EM console is accessible.
+   */
+  private void verifyDomainReady() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:288]: Can you embed the content of the methods verifyDomainReady() to test Method for clarity.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java (hunk 1 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:1]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java (hunk 2 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:143]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java (hunk 3 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:144]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java (hunk 4 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:160]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java (hunk 5 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:288]: I prefer a helper method to have the test method clean and simple, if you are ok about it

Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java:288]: Helper method is needed when we use repetitive code. Here I see one instance of the method  verifyDomainReady() in the class

**Review by anpanigr — COMMENTED** [review]
Update PR description with Jenkin Results

**Review by anpanigr — APPROVED** [review]
lgtm

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicClusterMiiDomain.java [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test to creat a FMW dynamic domain in model in image.
+ */
+@DisplayName("Test to Creat a FMW Dynamic Domain with Dynamic Cluster using model in image")
```
Comment (jshum2479) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicClusterMiiDomain.java:66]: Creat - Create

Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicClusterMiiDomain.java:66]: changed

**Review by jshum2479 — APPROVED** [review]

**Pull request merged**